File: customer_support_phoenix/lib/customer_support_phoenix/chat/chat_message.ex
defmodule CustomerSupportPhoenix.Chat.ChatMessage do
  use Ecto.Schema
  import Ecto.Changeset

  alias CustomerSupportPhoenix.Chat.{ChatSession, ChatMessageFeedback}

  schema "chat_messages" do
    field :sender, :string
    field :message, :string
    belongs_to :chat_session, ChatSession
    has_one :feedback, ChatMessageFeedback

    timestamps(type: :utc_datetime)
  end

  @doc false
  def changeset(chat_message, attrs) do
    chat_message
    |> cast(attrs, [:sender, :message, :chat_session_id])
    |> validate_required([:sender, :message, :chat_session_id])
  end
end

File: customer-support-laravel/app/Models/ChatMessage.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasOne;

class ChatMessage extends Model
{
    use HasFactory;

    protected $fillable = [
        'chat_session_id',
        'sender',
        'message',
    ];

    public function chatSession(): BelongsTo
    {
        return $this->belongsTo(ChatSession::class);
    }

    public function feedback(): HasOne
    {
        return $this->hasOne(ChatFeedback::class);
    }
}
